cpp/src/neighbors/detail/cagra/cagra_build.cuh

```diff
@@ -357,8 +353,7 @@ void build_knn_graph(
   raft::host_matrix_view<IdxT, int64_t, raft::row_major> knn_graph,
   cuvs::neighbors::nn_descent::index_params build_params)
 {
-  auto nn_descent_idx = cuvs::neighbors::nn_descent::index<IdxT>(res, knn_graph);
-  cuvs::neighbors::nn_descent::build<DataT, IdxT>(res, build_params, dataset, nn_descent_idx);
+  auto nn_descent_idx = cuvs::neighbors::nn_descent::build(res, build_params, dataset);
```

Comment:
This is relevant: https://github.com/rapidsai/cuvs/pull/264#discussion_r1778641188